### DISCUSSION TITLE
Update from v0.4.0 to v0.5.0

### DISCUSSION THREAD

**Discussion by luan-xiaokun** [eclipse-langium/langium#724] [Q&A] [answered]
Glad to see version 0.5 of Langium, thank you for this good  work, I can't wait to update the langium used in my project to version 0.5.

I changed the version requirements in `package.json` and ran `npm install`. The update went well, but the compilation reported many errors, most of which were related to `computeScope`.

What is the best way to update langium? Should I update the Yeoman langium generator to create a new project and copy the `.langium` file, or it is better to manually fix those errors?

**Comment by msujew** [comment, marked as answer]
Hey @luan-xiaokun,

We try to perform as little breaking changes as possible. However, for 0.5 we had to rename/move around a few services and methods for the sake of clearity. Most breaking changes (including `computeScope`) are documented in [our changelog](https://github.com/langium/langium/blob/main/packages/langium/CHANGELOG.md#breaking-changes). I would recommend going through that. The yeoman generator uses relatively little of the Langium API surface, and therefore might not be enough to accurately migrate to a new update.

**Reply by luan-xiaokun** [reply]
Thanks for your timely help! I will first go through the changelog (It turns out that I was looking in the wrong place for the changelog)
